Sources/StreamChat/StateLayer/MessageSearchState+Observer.swift

```diff
@@ -45,9 +45,8 @@ extension MessageSearchState {
                 )
                 do {
                     if let messagesObserver {
-                        try messagesObserver.startObserving(didChange: handlers.messagesDidChange)
-                        // Sending the initial value since we keep recreating the observer
-                        Task { await handlers.messagesDidChange(messagesObserver.items) }
+                        let messages = try messagesObserver.startObserving(didChange: handlers.messagesDidChange)
+                        Task.mainActor { await handlers.messagesDidChange(messages) }
```

Comment:
Special case where we keep recreating the observer whenever the search term changes